docker/Dockerfile

```diff
@@ -0,0 +1,49 @@
+FROM rust:1.39 as build
```

Comment:
This whole directory should go under `extras/`.

Reply:
Or maybe `tools/` is better...

Reply:
Yes I thought both `extras` and `tools` are not a good directory to use so simply created under `docker/`. If it needs to go into some subdirectory for cleanness, I prefer `extras/docker`.